packages/apps/plugins/plugin-status-bar/src/components/FeedbackForm.tsx

```diff
@@ -0,0 +1,133 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { X, PaperPlaneTilt } from '@phosphor-icons/react';
+import React, { type FC, type PropsWithChildren, useCallback } from 'react';
+
+import { useIntentDispatcher } from '@dxos/app-framework';
+import { S } from '@dxos/echo-schema';
+import { Button, Input, Popover, useTranslation } from '@dxos/react-ui';
+import { getSize, mx } from '@dxos/react-ui-theme';
+
+import { useForm } from '../hooks';
+import { STATUS_BAR_PLUGIN } from '../meta';
+import { mkTranslation } from '../translations';
+import { validate } from '../util';
+
+const Email = S.string.pipe(
+  S.nonEmpty({ message: () => 'Email is required.' }),
+  S.pattern(/^(?!\.)(?!.*\.\.)([A-Z0-9_+-.]*)[A-Z0-9_+-]@([A-Z0-9][A-Z0-9-]*\.)+[A-Z]{2,}$/i, {
+    message: () => 'Invalid email address.',
+  }),
+);
+
+const nonEmpty = (field: string) => S.nonEmpty({ message: () => `${field} is required.` });
+const maxLength = (field: string, length: number) => S.maxLength(length, { message: () => `${field} is too long.` });
+
+// -- Types and validation.
+const FeedbackFormSchema = S.struct({
+  name: S.string.pipe(nonEmpty('Name'), maxLength('Name', 256)),
+  email: Email.pipe(maxLength('Email', 256)),
+  message: S.string.pipe(nonEmpty('Feedback'), maxLength('Feedback', 32_768)),
+});
+
+type FeedbackFormState = S.Schema.Type<typeof FeedbackFormSchema>;
+
+const initialValues: FeedbackFormState = { name: '', email: '', message: '' };
+
+const Section: FC<PropsWithChildren & { className?: string }> = ({ children, className }) => (
+  <div role='none' className={mx(className)}>
+    {children}
+  </div>
+);
```

Comment:
Essentially equivalent to a literal `<div role='none'>` except it prevents refs or other attributes even though the type accepts them… I think it could be removed and consumers could just use the literal, or this could render a [`<fieldset>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) or provide `role='group'`…

Reply:
Otherwise, either its type should be narrowed to `PropsWithChildren<ThemedClassName<{}>>`, or it should spread its props (and possibly forward refs).

Reply:
As a general rule, the prop that forwards to `mx` should be called `classNames` (with an ‘s’), since in theory `className` should override anything here.

Reply:
You're right!